### PR TITLE
Fixed off-by-one error, restructured unit class.

### PR DESCRIPTION
**Pull request by admoore0**
Here you go kyle

### CODE REVIEW

File: src/cards/attack/attack_cards/swordsman_1.tres [hunk 1]
```diff
@@ -7,7 +7,7 @@
 script = ExtResource("1_h7uhr")
 attack_power = 4
 attack_recoil = 1
-health = 5
+health = 4
```
Comment (kdreese) [inline src/cards/attack/attack_cards/swordsman_1.tres:10]: We want to revert this back?

File: src/states/game/game.gd [hunk 2]
```diff
@@ -49,6 +49,7 @@ var game_over := false
 
 
 func _ready() -> void:
+	seed(0)
```
Comment (kdreese) [inline src/states/game/game.gd]: This one should be undone.

File: src/units/melee_unit.gd [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+
+var attack_power: int
+var recoil: int
+var health: int
+var max_health: int
+var speed: int
+
+
+func init(data: MeleeUnitData, lane: int) -> void:
```
Comment (thearst3rd) [inline src/units/melee_unit.gd:17]: TODO: change `lane` to `grid_position`

File: src/units/ranged_unit.gd [hunk 4]
```diff
@@ -8,20 +8,19 @@ extends Node2D
...
+	grid_position = starting_position
 	sprite.texture = data.icon
 	rank_icon.texture = Util.rank_to_texture(data.rank)
 	update_position()
 
 
 func update_position() -> void:
-	if row < 3:
-		position = Vector2(580, 40) + row * Vector2(0, 40)
+	if grid_position.y < 2:
```
Comment (kdreese) [inline src/units/ranged_unit.gd]: Oop

**Review by thearst3rd — APPROVED** [review]
k

File: src/cards/attack/attack_cards/swordsman_1.tres (hunk 1 above)
Comment (admoore0) [reply src/cards/attack/attack_cards/swordsman_1.tres:10]: Will hit when we rebalance everything later

**Review by kdreese — APPROVED** [review]

**Review by thearst3rd — APPROVED** [review]
fr this time

**Pull request merged**